### PR TITLE
EVG-7305 make creating a project from API update scopes

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: rest/data/project.go [hunk 1]
```diff
@@ -38,8 +39,8 @@ func (pc *DBProjectConnector) FindProjectById(id string) (*model.ProjectRef, err
 }
 
 // CreateProject inserts the given model.ProjectRef.
-func (pc *DBProjectConnector) CreateProject(projectRef *model.ProjectRef) error {
-	err := projectRef.Insert()
+func (pc *DBProjectConnector) CreateProject(projectRef *model.ProjectRef, u *user.DBUser) error {
+	err := projectRef.Add(u)
```
Comment (bsamek) [inline rest/data/project.go]: It's a little odd to me that a function with the signature `*ProjectRef.Add(*user.DbUser)` also inserts the *ProjectRef. Should these be separate functions?

File: rest/route/project.go [hunk 2]
```diff
@@ -534,7 +534,8 @@ func (h *projectIDPutHandler) Run(ctx context.Context) gimlet.Responder {
 	if err = responder.SetStatus(http.StatusCreated); err != nil {
 		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "Cannot set HTTP status code to %d", http.StatusCreated))
 	}
-	if err = h.sc.CreateProject(dbProjectRef); err != nil {
+	u := gimlet.GetUser(ctx).(*user.DBUser)
+	if err = h.sc.CreateProject(dbProjectRef, u); err != nil {
```
Comment (bsamek) [inline rest/route/project.go]: What happens if `u` is nil?

File: rest/data/project.go (hunk 1 above)
Comment (john-m-liu) [reply rest/data/project.go]: I don't think so - this combines insert with updating permissions. I feel like calling it Add implies that, but I can rename it to something like CreateProject

File: rest/route/project.go (hunk 2 above)
Comment (john-m-liu) [reply rest/route/project.go]: The line above panics, but there is already a check for user not being nil in the middleware

File: rest/data/project.go (hunk 1 above)
Comment (bsamek) [reply rest/data/project.go]: No need, SGTM.

**Pull request merged**